usecase/syncstrategy/window.go

```diff
@@ -46,42 +46,35 @@ func (window *Window) Sync(
 	})
 	logger.Debug("spawning goroutines for sync block workers")
 
+	workersErrGroup := errgroup.Group{}
+	commandWindowMutex := sync.Mutex{}
+
+	commandWindow := newUnsafeCommandWindow(beginHeight, endHeight)
+
 	for height := beginHeight; height <= endHeight; height += 1 {
-		go func(height int64) {
+		height := height
+		workersErrGroup.Go(func() error {
 			commands, err := worker(height)
 			if err != nil {
-				workResultCh <- workResult{height, nil, err}
+				logger.Errorf("received error from sync block worker #%d: %v", height, err)
+				return err
 			}
 
-			workResultCh <- workResult{height, commands, nil}
-		}(height)
-	}
+			result := workResult{height, commands, nil}
```

Comment:
I think we no longer need the error to be stored here.

Reply:
Correct, removed the `workResult` struct